tests/integration/java/com/github/bazel_contrib/target_determinator/integration/Tests.java

```diff
@@ -479,6 +480,11 @@ public void testChmodFile() throws TargetComputationErrorException {
     doTest(Commits.ONE_SH_TEST, Commits.SH_TEST_NOT_EXECUTABLE, Set.of("//sh:sh_test"));
   }
 
+  @Test
+  public void incompatibleTargetsAreFiltered() throws TargetComputationErrorException {
+    doTest(Commits.NO_TARGETS, Commits.INCOMPATIBLE_TARGET, Set.of("//java/example:CompatibleTest"));
```

Comment:
Is NO_TARGETS a good base revision to use? Looks like bazel-dffer does not handle the no-target case. May be I should have added CompatibleTest and IncompatibleTest targets in two different commits. WDYT?

Reply:
Ah lovely - let's maybe use `ONE_TEST` instead - deleted targets shouldn't get flagged as affected, and this means the other TDs will work (even though it's a bit annoying to need to!)

Reply:
Also, can you explicitly pass a set containing the incompatible target as the fourth arg to this function, as this 's not just an unexpected but actively problematic target to return?

Finally, if (as I'd expect) the other TDs _do_ return the incompatible target, can you add overrides for the new test in their specific subclasses with a comment explaining that they're broken?

Thanks for putting all this together!

Reply:
Done, PTAL